### ISSUE TITLE
SA1600 triggers for `protected internal`

### ISSUE THREAD

**Issue by smaillet** [DotNetAnalyzers/StyleCopAnalyzers#3950]
SA1600 is triggering for methods marked as `protected internal`.

``` C#
protected internal void ShowBogusError()
{
}
```

This is not a public facing method, it is BOTH `protected` AND `internal` so no error should appear. (You are at least in good company with this bug as [CS1591](https://github.com/dotnet/roslyn/issues/80822) has the same problem)

**Comment by bjornhellander** [comment]
See https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers:
_Only code in the same assembly or in a derived class in another assembly can access this type or member._

This behavior seeems to be correct.

**Comment by manfred-brands** [comment]
@smaillet You would need to use `private protected` to get what you want.

_Only code in the same assembly and in the same class or a derived class can access the type or member._

**Comment by sharwell** [comment]
@bjornhellander is correct here. `protected internal` means protected OR internal, while `private protected` means protected AND internal.
